src/presets/cloudflare/utils.ts

```diff
@@ -182,37 +186,116 @@ async function writeCFPagesRedirects(nitro: Nitro) {
     contents.unshift(currentRedirects);
   }
 
-  await fsp.writeFile(redirectsPath, contents.join("\n"));
+  await writeFile(redirectsPath, contents.join("\n"), true);
 }
 
-async function writeCFWrangler(nitro: Nitro) {
-  type WranglerConfig = typeof nitro.options.cloudflare.wrangler;
+async function writeCFWranglerConfig(nitro: Nitro) {
+  const extraConfig: WranglerConfig = nitro.options.cloudflare?.wrangler || {};
 
-  const inlineConfig: WranglerConfig =
-    nitro.options.cloudflare?.wrangler || ({} as WranglerConfig);
-
-  // Write wrangler.toml only if config is not empty
-  if (!inlineConfig || Object.keys(inlineConfig).length === 0) {
+  // Skip if there are no extra config
+  if (Object.keys(extraConfig || {}).length === 0) {
     return;
   }
 
-  let configFromFile: WranglerConfig = {} as WranglerConfig;
-  const configPath = resolve(
-    nitro.options.rootDir,
-    inlineConfig.configPath || "wrangler.toml"
-  );
-  if (existsSync(configPath)) {
-    configFromFile = parseTOML<WranglerConfig>(
-      await fsp.readFile(configPath, "utf8")
+  // Read user config
+  const userConfig = await resolveWranglerConfig(nitro.options.rootDir);
+
+  // Merge configs
+  const mergedConfig = userConfig.config
+    ? mergeWranglerConfig(userConfig.config, extraConfig)
+    : extraConfig;
+
+  // Write config
+  // https://github.com/cloudflare/workers-sdk/pull/7442
+  const configRedirect = !!process.env.EXPERIMENTAL_WRANGLER_CONFIG;
+  if (configRedirect) {
+    if (mergedConfig.pages_build_output_dir) {
+      throw new Error(
+        "`pages_build_output_dir` wrangler config should not be set."
+      );
+    }
```

Comment:
Oh I see that you actively do not want the user to set this.
OK. But in that case you need to provide it automatically.

Reply:
Perhaps add after the check?

```
    mergedConfig.pages_build_output_dir = nitro.options.output.publicDir;
```